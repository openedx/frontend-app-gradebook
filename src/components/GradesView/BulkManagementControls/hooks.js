import { actions, selectors } from 'data/redux/hooks';

export const useBulkManagementControlsData = () => {
  const gradeExportUrl = selectors.root.useGradeExportUrl();
  const showBulkManagement = selectors.root.useShowBulkManagement();
  const downloadBulkGradesReport = actions.grades.useDownloadBulkGradesReport();

  const handleClickExportGrades = () => {
    downloadBulkGradesReport();
    window.location.assign(gradeExportUrl);
  };

  return {
    show: showBulkManagement,
    handleClickExportGrades,
  };
};
export default useBulkManagementControlsData;
