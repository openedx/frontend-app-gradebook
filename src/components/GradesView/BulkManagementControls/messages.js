import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  downloadGradesBtn: {
    id: 'gradebook.GradesView.BulkManagementControls.bulkManagementLabel',
    defaultMessage: 'Download Grades',
    description: 'A labeled button that allows an admin user to download course grades all at once (in bulk).',
  },
});

export default messages;
