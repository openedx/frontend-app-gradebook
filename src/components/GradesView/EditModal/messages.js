import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  assignmentHeader: {
    id: 'gradebook.GradesView.EditModal.headers.assignment',
    defaultMessage: 'Assignment',
    description: 'Edit Modal Assignment header',
  },
  currentGradeHeader: {
    id: 'gradebook.GradesView.EditModal.headers.currentGrade',
    defaultMessage: 'Current Grade',
    description: 'Edit Modal Current Grade header',
  },
  originalGradeHeader: {
    id: 'gradebook.GradesView.EditModal.headers.originalGrade',
    defaultMessage: 'Original Grade',
    description: 'Edit Modal Original Grade header',
  },
  studentHeader: {
    id: 'gradebook.GradesView.EditModal.headers.student',
    defaultMessage: 'Student',
    description: 'Edit Modal Student header',
  },
  title: {
    id: 'gradebook.GradesView.EditModal.title',
    defaultMessage: 'Edit Grades',
    description: 'Edit Modal title',
  },
  closeText: {
    id: 'gradebook.GradesView.EditModal.closeText',
    defaultMessage: 'Cancel',
    description: 'Edit Modal close button text',
  },
  visibility: {
    id: 'gradebook.GradesView.EditModal.contactSupport',
    defaultMessage: 'Showing most recent actions (max 5). To see more, please contact support',
    description: 'Edit Modal visibility hint message',
  },
  saveVisibility: {
    id: 'gradebook.GradesView.EditModal.saveVisibility',
    defaultMessage: 'Note: Once you save, your changes will be visible to students.',
    description: 'Edit Modal saved changes effect hint',
  },
  saveGrade: {
    id: 'gradebook.GradesView.EditModal.saveGrade',
    defaultMessage: 'Save Grades',
    description: 'Edit Modal Save button label',
  },
});

export default messages;
