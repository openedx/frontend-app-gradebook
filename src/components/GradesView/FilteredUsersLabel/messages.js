import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  visibilityLabel: {
    id: 'gradebook.GradesTab.usersVisibilityLabel',
    defaultMessage: 'Showing {filteredUsers} of {totalUsers} total learners',
    description: 'Users visibility label',
  },
});

export default messages;
