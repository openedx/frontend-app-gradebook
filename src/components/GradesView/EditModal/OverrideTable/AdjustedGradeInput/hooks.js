import { actions, selectors } from 'data/redux/hooks';
import { getLocalizedSlash } from 'i18n/utils';

const useAdjustedGradeInputData = () => {
  const possibleGrade = selectors.root.useEditModalPossibleGrade();
  const value = selectors.app.useModalData().adjustedGradeValue;
  const setModalState = actions.app.useSetModalState();
  const hintText = possibleGrade && ` ${getLocalizedSlash()} ${possibleGrade}`;

  const onChange = ({ target }) => {
    setModalState({ adjustedGradeValue: target.value });
  };

  return {
    value,
    onChange,
    hintText,
  };
};

export default useAdjustedGradeInputData;
