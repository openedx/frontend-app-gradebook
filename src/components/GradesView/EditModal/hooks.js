import { selectors, actions, thunkActions } from 'data/redux/hooks';

export const useEditModalData = () => {
  const error = selectors.grades.useGradeData().gradeOverrideHistoryError;
  const isOpen = selectors.app.useModalData().open;
  const closeModal = actions.app.useCloseModal();
  const doneViewingAssignment = actions.grades.useDoneViewingAssignment();
  const updateGrades = thunkActions.grades.useUpdateGrades();

  const onClose = () => {
    doneViewingAssignment();
    closeModal();
  };

  const handleAdjustedGradeClick = () => {
    updateGrades();
    doneViewingAssignment();
    closeModal();
  };

  return {
    onClose,
    error,
    handleAdjustedGradeClick,
    isOpen,
  };
};

export default useEditModalData;
