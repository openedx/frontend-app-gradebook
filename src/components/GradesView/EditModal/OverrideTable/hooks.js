import { useIntl } from '@edx/frontend-platform/i18n';

import { gradeOverrideHistoryColumns as columns } from 'data/constants/app';
import { selectors } from 'data/redux/hooks';

import messages from './messages';

const useOverrideTableData = () => {
  const { formatMessage } = useIntl();

  const hide = selectors.grades.useHasOverrideErrors();
  const gradeOverrides = selectors.grades.useGradeData().gradeOverrideHistoryResults;
  const tableProps = {};
  if (!hide) {
    tableProps.columns = [
      { Header: formatMessage(messages.dateHeader), accessor: columns.date },
      { Header: formatMessage(messages.graderHeader), accessor: columns.grader },
      { Header: formatMessage(messages.reasonHeader), accessor: columns.reason },
      { Header: formatMessage(messages.adjustedGradeHeader), accessor: columns.adjustedGrade },
    ];
    tableProps.data = gradeOverrides;
  }
  return { hide, ...tableProps };
};

export default useOverrideTableData;
