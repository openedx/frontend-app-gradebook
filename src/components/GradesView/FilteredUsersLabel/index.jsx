import React from 'react';
import PropTypes from 'prop-types';

import { useIntl } from '@edx/frontend-platform/i18n';

import { selectors } from 'data/redux/hooks';
import messages from './messages';

export const BoldText = ({ text }) => (
  <span className="font-weight-bold">{text}</span>
);
BoldText.propTypes = {
  text: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
};

/**
 * <FilteredUsersLabel />
 * Simple label component displaying the filtered and total users shown
 */
export const FilteredUsersLabel = () => {
  const { filteredUsersCount, totalUsersCount } = selectors.grades.useUserCounts();
  const { formatMessage } = useIntl();

  if (!totalUsersCount) {
    return null;
  }
  return formatMessage(
    messages.visibilityLabel,
    {
      filteredUsers: <BoldText text={filteredUsersCount} />,
      totalUsers: <BoldText text={totalUsersCount} />,
    },
  );
};
FilteredUsersLabel.propTypes = {};

export default FilteredUsersLabel;
