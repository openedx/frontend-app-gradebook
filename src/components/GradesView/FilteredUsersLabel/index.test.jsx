import React from 'react';
import { shallow } from 'enzyme';

import { useIntl } from '@edx/frontend-platform/i18n';

import { formatMessage } from 'testUtils';
import { selectors } from 'data/redux/hooks';

import FilteredUsersLabel, { BoldText } from '.';
import messages from './messages';

jest.mock('data/redux/hooks', () => ({
  selectors: {
    grades: {
      useUserCounts: jest.fn(),
    },
  },
}));

const userCounts = {
  filteredUsersCount: 100,
  totalUsersCount: 123,
};
selectors.grades.useUserCounts.mockReturnValue(userCounts);

let el;
describe('FilteredUsersLabel component', () => {
  beforeEach(() => {
    jest.clearAllMocks();
    el = shallow(<FilteredUsersLabel />);
  });
  describe('behavior', () => {
    it('initializes intl hook', () => {
      expect(useIntl).toHaveBeenCalled();
    });
    it('initializes redux hooks', () => {
      expect(selectors.grades.useUserCounts).toHaveBeenCalled();
    });
  });
  describe('render', () => {
    test('null render if totalUsersCount is 0', () => {
      selectors.grades.useUserCounts.mockReturnValueOnce({
        ...userCounts,
        totalUsersCount: 0,
      });
      expect(shallow(<FilteredUsersLabel />).isEmptyRender()).toEqual(true);
    });
    test('snapshot', () => {
      expect(el).toMatchSnapshot();
      expect(el).toMatchObject(shallow(formatMessage(messages.visibilityLabel, {
        filteredUsers: <BoldText text={userCounts.filteredUsersCount} />,
        totalUsers: <BoldText text={userCounts.totalUsersCount} />,
      })));
    });
  });
});
