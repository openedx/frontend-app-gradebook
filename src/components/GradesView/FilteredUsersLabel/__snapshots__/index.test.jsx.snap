// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilteredUsersLabel component render snapshot 1`] = `
<format-message-function
  message={
    Object {
      "defaultMessage": "Showing {filteredUsers} of {totalUsers} total learners",
      "description": "Users visibility label",
      "id": "gradebook.GradesTab.usersVisibilityLabel",
    }
  }
  values={
    Object {
      "filteredUsers": <BoldText
        text={100}
      />,
      "totalUsers": <BoldText
        text={123}
      />,
    }
  }
/>
`;
